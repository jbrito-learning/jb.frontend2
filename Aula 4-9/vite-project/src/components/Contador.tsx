import { useState } from "react";

const Contador = () => {
  const [counter, setCounter] = useState(0);

  return (
    <div>
      <div>Counter</div>
      <div>Valor: {counter}</div>
      <button onClick={() => setCounter(counter + 1)}>Incrementar</button>
    </div>
  );
};

export default Contador;
