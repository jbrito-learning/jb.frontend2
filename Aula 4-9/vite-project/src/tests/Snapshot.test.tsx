import { render } from "@testing-library/react";
import Contador from "../components/Contador";

test("Must match snapshot", () => {
  const { container } = render(<Contador />);
  expect(container).toMatchSnapshot();
});
