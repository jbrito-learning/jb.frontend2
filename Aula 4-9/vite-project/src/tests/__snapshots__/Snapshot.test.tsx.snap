// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must match snapshot 1`] = `
<div>
  <div>
    <div>
      Counter
    </div>
    <div>
      Valor: 
      0
    </div>
    <button>
      Incrementar
    </button>
  </div>
</div>
`;
